src/main/java/duke/ui/Ui.java

```diff
@@ -0,0 +1,93 @@
+package duke.ui;
+
+import duke.task.Task;
+import duke.task.TaskList;
+
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.util.Scanner;
+
+public class Ui {
+    private static final String INDENTATION = "    ";
+    private static final String BORDER = INDENTATION + "--------------------------------"
+            + "--------------------------------------";
+    private static final String LOGO = " ____        _        \n"
+            + "|  _ \\ _   _| | _____ \n"
+            + "| | | | | | | |/ / _ \\\n"
+            + "| |_| | |_| |   <  __/\n"
+            + "|____/ \\__,_|_|\\_\\___|\n";
+    private static final String ADD_TASK_ACK = "Got it. I've added this task:";
+
+
+
+    private final Scanner sc;
+    private final PrintStream out;
+
+    public Ui(InputStream in, PrintStream out) {
+        this.sc = new Scanner(in);
+        this.out = out;
+    }
+
+    public Ui() {
+        this(System.in, System.out);
+    }
+
+    // Varargs used here
+    public void display(String... msg) {
+        out.println(BORDER);
+        for (String m : msg) {
+            out.println(INDENTATION + m);
+        }
+        out.println(BORDER);
+    }
+
+    public void showGreeting() {
+        display("Hello from \n" + LOGO, "How may I help you?");
+    }
+
+    public String readCommand() {
+        return sc.nextLine().trim();
+    }
+
+    public void showError(String msg) {
+        display(msg);
+    }
+
+    public void printAddTaskAck(Task t, TaskList tasks) {
+        String taskCountMsg = "You now have " + tasks.getTasksSize() + " task(s) in the list.";
+        display(ADD_TASK_ACK, " " + t.toString(), taskCountMsg);
+    }
+
+    public void printExitMsg() {
```

Comment:
Same for the ones over here. Would be good if you use a consistent verb (between `print`, `display` and `show`)